base/ltplain.dtx

```diff
@@ -1186,18 +1186,143 @@
+%    put (because it was equal to to the primitive \cs{par}), now
+%    \cs{obeyedline} expands and you get what it  contains, i.e., in that
+%    case \cs{par}, into the \cs{edef} or \cs{mark} unless we use
+%    \cs{protected} on it.
+%    \begin{macrocode}
+\protected\gdef\obeyedline{\par}
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\obeyedspace}
+%    The definition of \cs{obeyspaces} is changed in the same way and
+%    now executes \cs{obeyedspace} for each active space.
 %    \begin{macrocode}
-{\catcode`\^^M=\active % these lines must end with %
-  \gdef\obeylines{\catcode`\^^M\active \let^^M\par}%
-  \global\let^^M\par} % this is in case ^^M appears in a \write
-\def\obeyspaces{\catcode`\ \active}
-{\obeyspaces\global\let =\space}
+\global\let\obeyedspace\space
 %    \end{macrocode}
 % \end{macro}
+%
+%    \begin{macrocode}
+\begingroup
+\catcode`\ =\active%
+\global\let =\space%
+\gdef\obeyspaces{\catcode`\ \active\let =\obeyedspace}%
+\endgroup
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%</2ekernel|latexrelease>
+%<latexrelease>\EndIncludeInRelease
+%<latexrelease>\IncludeInRelease{0000/00/00}{\obeylines}%
+%<latexrelease>    {Add a redirection to obeylines and obeyspaces}%
+%<latexrelease>
+%    \end{macrocode}
+%    From 2019 onwards the commands are made robust (somewhat later in
+%    the kernel sources). So if we roll back they are robust, so when
+%    redefining them we have get rid of the robust payload
+%    first. Otherwise that is seen by the later rollback below, which
+%    then installs a fragile version of the new definition on top of
+%    the one we roll back to here, sigh. \cs{kernel@make@fragile} also
+%    changes its definition (later own) so this is done directly.
+%    \begin{macrocode}
+%<latexrelease>\expandafter\let\csname obeylines \endcsname\@undefined
+%<latexrelease>\expandafter\let\csname obeyspace \endcsname\@undefined
+%<latexrelease>
+%<latexrelease>\begingroup
+%<latexrelease>\catcode`\^^M=\active % these lines must end with %
+%<latexrelease>  \gdef\obeylines{\catcode`\^^M\active \let^^M\par %
+%    \end{macrocode}
+%    Closing brace on a separate line (see comment above).
+%    \begin{macrocode}
+%<latexrelease>                 }%
+%    \end{macrocode}
+%    Another pitfall: if we do a rollback \cs{par} is no longer the
+%    primitive, so the roll back definition needs \cs{let} to what is
+%    new the primitive.
+%    \begin{macrocode}
+%<latexrelease>  \global\let^^M\RawParEnd % this is in case ^^M appears in a \write
+%<latexrelease>\endgroup
+%<latexrelease>\def\obeyspaces{\catcode`\ \active}
+%<latexrelease>
+%<latexrelease>\let\obeyedline\@undefined
+%<latexrelease>\let\obeyedspace\@undefined
+%<latexrelease>\EndIncludeInRelease
+%<*2ekernel>
+%    \end{macrocode}
+%
+%    An active space generates \cs{space} by default (for example in a \cs{write}):  
+%    \begin{macrocode}
+%    \end{macrocode}
```

Comment:
Stray lines?

Reply:
well, text partly but not fully reordered